### PR TITLE
Updates after release 3.0

### PR DESCRIPTION
**Pull request by samtroll**

### CODE REVIEW

**Comment by asbjornu** [comment]
I made a local branch of this PR in #26 to see whether the integration tests succeed, as they can't run on pull requests due to secrets not being exposed to pull requests.

**Comment by asbjornu** [comment]
@samtroll, can you please respond to the questions posed by @aait?

File: src/PayEx/Api/Client/Client.php [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+     * @throws Exception
+     */
+    public function request($requestMethod, $requestEndpoint, $requestParams = [])
+    {
+        $this->setMethod($requestMethod);
+        $this->setEndpoint($requestEndpoint);
+        $this->setParams($requestParams);
+        $requestBody = '';
+
+        if ($this->getMethod() == "GET") {
```
[the remark this answers is not in the document]
Comment (samtroll) [reply src/PayEx/Api/Client/Client.php]: The client library has been developed according to the PayEx API specifications at:
https://developer.payex.com/xwiki/wiki/developer/view/Main/ecommerce/technical-reference/

Nowhere is the DELETE method used.

File: src/PayEx/Api/Client/Client.php [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+            $this->setHeaders();
+            $this->setEndpoint(
+                $this->generateRequestURI(
+                    $this->getEndpoint(),
+                    $this->getParams()
+                )
+            );
+        }
+
+        if ($this->getMethod() != "GET") {
```
[the remark this answers is not in the document]
Comment (samtroll) [reply src/PayEx/Api/Client/Client.php]: The client library has been developed according to the PayEx API specifications at:
https://developer.payex.com/xwiki/wiki/developer/view/Main/ecommerce/technical-reference/

Nowhere is the DELETE method used.

File: src/PayEx/Api/Service/Consumer/Resource/ConsumerNationalIdentifier.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        return $this->offsetGet(self::COUNTRY_CODE);
+    }
+
+    /**
+     * @param string $countryCode
+     * @return $this
+     */
+    public function setCountryCode($countryCode)
+    {
+        return $this->offsetSet(self::COUNTRY_CODE, $countryCode);
```
[the remark this answers is not in the document]
Comment (samtroll) [reply src/PayEx/Api/Service/Consumer/Resource/ConsumerNationalIdentifier.php]: The resource classes are just data transfer objects, any validation or other business logic should be the responsibility of the model implementing it (assuming the MVC design pattern).

File: tests/ConsumerTest.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * @var ResponseInterface $consumerSessionService
+     */
+    protected $consumerSessionService;
+
+    /**
+     * @param bool $noTest
+     * @throws \PayEx\Api\Client\Exception
+     */
+    public function testInitiateConsumerSession($noTest = false)
```
[the remark this answers is not in the document]
Comment (samtroll) [reply tests/ConsumerTest.php]: It's a flag for enabling the initiation of a consumer session without running tests. This to make it reusable for testing other parts of the checkin process without repeating the same tests.

File: tests/ConsumerTest.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+            $consumerSessionData->setNationalIdentifier($nationalIdentifierData);
+
+            $consumerSessionRequest = new InitiateConsumerSession($consumerSessionData);
+            $consumerSessionRequest->setClient($this->client);
+
+            $this->consumerSessionService = $consumerSessionRequest->send();
+        }
+
+        if ($noTest) {
```
[the remark this answers is not in the document]
Comment (samtroll) [reply tests/ConsumerTest.php]: It's a flag for enabling the initiation of a consumer session without running tests. This to make it reusable for testing other parts of the checkin process without repeating the same tests.

File: src/PayEx/Api/Service/Request.php [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+    /**
+     * @return ResponseInterface
+     * @throws ClientException
+     */
+    public function send()
+    {
+        if (!$this->getClient()->getMerchantToken() ||
+            $this->getClient()->getMerchantToken() === '<merchant_token>') {
+            throw new ClientException('MERCHANT_TOKEN not configured in INI file or environment variable.');
```
[the remark this answers is not in the document]
Comment (samtroll) [reply src/PayEx/Api/Service/Request.php]: This is a remnant from old version of the library (2.x). We use environment-variables in our bitbucket pipeline tests and so does Travis CI if I understand it correctly, @asbjornu?

File: src/PayEx/Api/Service/Request.php [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+        }
+
+        if (!$this->getClient()->getPayeeId() ||
+            $this->getClient()->getPayeeId() === '<payee_id>') {
+            throw new ClientException('PAYEE_ID not configured in INI file or environment variable.');
```
[the remark this answers is not in the document]
Comment (samtroll) [reply src/PayEx/Api/Service/Request.php]: Also remnant from old version...

File: src/PayEx/Api/Service/Request.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (asbjornu) [reply src/PayEx/Api/Service/Request.php]: Yes, I suppose this is more for local testing where the creation of an `.ini` file is easier than setting environment variables? I think that supporting both `.ini` and environment variables is fine.

**Pull request merged**